librz/core/cbin.c

```diff
@@ -2946,6 +2946,36 @@ RZ_API bool rz_core_bin_whole_strings_print(RZ_NONNULL RzCore *core, RZ_NONNULL
 	return res;
 }
 
+RZ_IPI bool rz_core_bin_xrefs_strings_print(RZ_NONNULL RzCore *core, RZ_NONNULL RzBinFile *bf, RZ_NONNULL RzCmdStateOutput *state) {
+	rz_return_val_if_fail(core && state, false);
+
+	RzPVector *l = rz_core_bin_whole_strings(core, bf);
+	RzPVector *ll = rz_pvector_new(NULL);
+
+	void **iter;
+	rz_pvector_foreach (l, iter) {
+		RzBinString *string = *iter;
+		RzList *list = rz_analysis_xrefs_get_to(core->analysis, string->paddr);
+		RzAnalysisXRef *xref;
+		RzListIter *iterator;
+		rz_list_foreach (list, iterator, xref) {
+			switch (xref->type) {
+			case RZ_ANALYSIS_XREF_TYPE_CODE:
+			case RZ_ANALYSIS_XREF_TYPE_DATA:
+				if (!rz_pvector_contains(ll, string)) {
+					rz_pvector_push_front(ll, string);
```

Comment:
The string needs to be `poped` from the other list, otherwise there are two owners and you get a double free.